compiler/semmagic.nim

```diff
@@ -305,7 +305,7 @@ proc semOf(c: PContext, n: PNode): PNode =
         result.info = n.info
         result.typ = getSysType(c.graph, n.info, tyBool)
         return result
-      elif diff == high(int):
+      elif diff == high(int) and commonSuperclass(a, b) == nil:
```

Comment:
But that's what ``diff == high(int)`` means, no?

Reply:
Not really, suppose you have ``Base`` class and two children``A`` and ``B``. Now you are calling: ``A of B``. In this case ``inheritanceDiff()`` returns ``high(int)``, but ``commonSuperclass()`` returns ``Base``. When``A of B`` is called, I don't want compilation error because in my case it is result of template expansion (very frequiently used template ``log``). I want to get simply false.

Reply:
Meh, ok.